src/domain/worklets/AudioInputProcessor.ts

```diff
@@ -58,10 +59,17 @@ class AudioInputProcessor extends AudioWorkletProcessor {
     _processor: ((input: Array<Float32Array>) => void);
     _haveReportedUpSampleError = false;
 
+    _ringBufferStorage: SharedArrayBuffer;
+    _ringBuffer: RingBuffer<Int16Array>;
+
 
     constructor(options?: AudioWorkletNodeOptions) {
         super(options);  // eslint-disable-line
 
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+        this._ringBufferStorage = options?.processorOptions?.ringBufferStorage as SharedArrayBuffer;
```

Comment:
This line is not necessarily type safe. Do we need to account for a situation where `options`, `processorOptions`, or `ringBufferStorage` are undefined?

Reply:
Not really, that would be a programmer error and should fail loudly cause there is sort of a type check in the `RingBuffer` constructor. Ideally we would have options parameter typed to require this, but the call site is not direct in this case, It happens in the `AudioInput.ts` through instantiation of `AudioWorkletNode` with an string id that is associated with `AudioInputProcessor` in `AudioWorklets.ts`. In theory typescript should be able to handle all of that, I think I saw a feature that allows mapping string literals to a types, but it might get tricky if the browser builtin typing we're using can't accommodate something like that.

Reply:
Ow no, the "sort of type check" is for another parameter, so I'll add explicit check there with an exception.